### PR TITLE
Refactor long <h4> in user guide to bold text

### PR DESCRIPTION
**Pull request by Chng-Zhi-Xuan**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Documentation update

Part of #335 

**What is the rationale for this request?**
Heading font sizes on mobile site was too large.

**What changes did you make? (Give an overview)**
- Change long instances of `<h4>` into bold markdown text using `**`

Before | After 
--- | ---
335-font-size-before | 335-font-size-afterv2

**Is there anything you'd like reviewers to focus on?**
- Readability of new format on mobile.

**Testing instructions:**
- Use `Netlify Preview` and reduce your window size.
- Go through the document and check the readability of the content.

### CODE REVIEW

**Comment by yamgent** [comment]
Ok. However, I took a look at this with regards to `website-base`, and didn't think that much space was saved with the reduction in heading. And for h4 and below (see second row and third row below), we can no longer differentiate between the headings and the normal text because the sizes are almost the same.

To me, I think both "Before" and "After" are acceptable, but this is probably down to personal taste. @acjh what do you think about it?

Before | After
-- | --
 | 
 | 
 | 

Url of the pages shown above:
https://nus-cs2103.github.io/website-base/admin/programming-languages.html
https://nus-cs2103.github.io/website-base/se-book-adapted/chapters/ides.html

File: asset/css/markbind.css [hunk 1]
```diff
@@ -33,6 +33,32 @@ code {
 
 /* Bootstrap small(sm) responsive breakpoint */
 @media (max-width: 767.98px) {
+
+    /* 70% of Bootstrap 4 default font-size for headings */
+    h1 {
```
Comment (yamgent) [inline asset/css/markbind.css]: Bootstrap also overrides classes of `h1` (i.e. it does `h1, .h1 {`). Should we consider doing the same?

File: docs/userGuide/components/modal.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Modal
 
-#### Modals are to be used together with the [Trigger](#trigger) component for activation.
+<p><strong>Modals are to be used together with the <md>[Trigger](#trigger)</md> component for activation.</strong></p>
```
Comment (yamgent) [inline docs/userGuide/components/modal.md]: Can just use standard Markdown syntax, it renders the same way:

```
**Modals are to be used together with the [Trigger](#trigger) component for activation.**
```

**Comment by Chng-Zhi-Xuan** [comment]
Bold text was intended to substitute `<h4>`. Long blocks of text should remain as is for easier readability (especially on mobile). 

Will just have that paragraph use the default responsive font size from Bootstrap.

File: docs/userGuide/components/advanced.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## Rich formatted text in headings and titles
 
-<p style="font-size:18px;">
+<p>
```
Comment (yamgent) [inline docs/userGuide/components/advanced.md]: The paragraph tag can just be removed then.

**Pull request merged**